lib/models/service.ts

```diff
@@ -0,0 +1,165 @@
+  readonly serviceId: NonEmptyString;
+  // the name of the department within the service
+  readonly departmentName: NonEmptyString;
+  // the name of the service
+  readonly serviceName: NonEmptyString;
+  // the name of the organization
+  readonly organizationName: NonEmptyString;
+  // list of authorized fiscal codes
+  readonly authorizedRecipients?: ReadonlyArray<FiscalCode>;
+  // user's subscription linked to this service
+  readonly subscriptionId: NonEmptyString;
+}
+
+/**
+ * Interface for new Service objects
+ */
+export interface INewService
+  extends IService,
+    DocumentDb.NewDocument,
+    IVersionedModel {
+  readonly kind: "INewService";
+}
+
+/**
+ * Interface for retrieved Service objects
+ *
+ * Existing Service records have a version number.
+ */
+export interface IRetrievedService
+  extends IService,
+    DocumentDb.RetrievedDocument,
+    IVersionedModel {
+  readonly id: NonEmptyString;
+  readonly kind: "IRetrievedService";
+}
+
+function toRetrieved(result: DocumentDb.RetrievedDocument): IRetrievedService {
+  return {
+    ...result,
+    kind: "IRetrievedService"
+  } as IRetrievedService;
+}
+
+function getModelId(o: IService): ModelId {
+  return nonEmptyStringToModelId(o.serviceId);
+}
+
+function updateModelId(
+  o: IService,
+  id: string,
+  version: NonNegativeNumber
+): INewService {
+  const newService: INewService = {
+    ...o,
+    id,
+    kind: "INewService",
+    version
+  };
+  return newService;
+}
+
+function toBaseType(o: IRetrievedService): IService {
+  return {
+    departmentName: o.departmentName,
+    organizationName: o.organizationName,
+    serviceId: o.serviceId,
+    serviceName: o.serviceName,
+    subscriptionId: o.subscriptionId
+  };
+}
+
+/**
+ * A model for handling Services
+ */
+export class ServiceModel extends DocumentDbModelVersioned<
+  IService,
+  INewService,
+  IRetrievedService
+> {
+  protected dbClient: DocumentDb.DocumentClient;
+  protected collectionUri: DocumentDbUtils.IDocumentDbCollectionUri;
+
+  /**
+   * Creates a new Service model
+   *
+   * @param dbClient the DocumentDB client
+   * @param collectionUrl the collection URL
+   */
+  constructor(
+    dbClient: DocumentDb.DocumentClient,
+    collectionUrl: DocumentDbUtils.IDocumentDbCollectionUri
+  ) {
+    super();
+    // tslint:disable-next-line:no-object-mutation
+    this.toRetrieved = toRetrieved;
+    // tslint:disable-next-line:no-object-mutation
+    this.getModelId = getModelId;
+    // tslint:disable-next-line:no-object-mutation
+    this.versionateModel = updateModelId;
+    // tslint:disable-next-line:no-object-mutation
+    this.toBaseType = toBaseType;
+    // tslint:disable-next-line:no-object-mutation
+    this.dbClient = dbClient;
+    // tslint:disable-next-line:no-object-mutation
+    this.collectionUri = collectionUrl;
+  }
+
+  /**
+   * Searches for one Service associated to the provided ID
+   *
+   * @param fiscalCode
+   */
+  public findByServiceId(
```

Comment:
Why don't we just keep `serviceId` == `subscriptionId` and avoid a having two IDs?

Reply:
because I misundestood this comment:  "Unica modifica che farei è non usare subscriptionId per l'id, ma avere un campo azureSubscriptionId, così sleghiamo l'esistenza di Service dalla subscriptionId (che un domani possiamo eliminare completamente come attributo)." 

I will get rid of serviceId field

Reply:
I think that was my comment 😄 but looking at the code now (and the issue with the multi partition query), I would not agree with it anymore